app/lib/features/space/widgets/events_card.dart

```diff
@@ -1,20 +1,73 @@
-import 'package:acter/features/events/providers/events_provider.dart';
+import 'dart:math';
+
+import 'package:acter/common/utils/routes.dart';
+import 'package:acter/common/widgets/default_button.dart';
+import 'package:acter/features/events/providers/event_providers.dart';
+import 'package:acter/features/events/widgets/events_item.dart';
 import 'package:flutter/material.dart';
-import 'package:acter/common/providers/space_providers.dart';
-import 'package:acter/features/events/widgets/events_calendar.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
 
-// ignore: must_be_immutable
 class EventsCard extends ConsumerWidget {
   final String spaceId;
   const EventsCard({super.key, required this.spaceId});
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final space = ref.watch(spaceProvider(spaceId)).requireValue;
-    final events = ref.watch(spaceEventsProvider(space));
-    return EventsCalendar(
-      events: events,
+    final events = ref.watch(spaceEventsProvider(spaceId));
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: <Widget>[
+        Text(
+          'Events',
+          style: Theme.of(context).textTheme.titleMedium,
+        ),
+        events.when(
+          error: (error, stackTrace) => Text(
+            'Loading events failed: $error',
+          ),
+          data: (events) {
+            int eventsLimit = min(events.length, 3);
+            return Padding(
+              padding: const EdgeInsets.only(top: 5),
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  events.isNotEmpty
+                      ? ListView.builder(
+                          shrinkWrap: true,
+                          itemCount: eventsLimit,
+                          physics: const NeverScrollableScrollPhysics(),
+                          itemBuilder: (context, idx) =>
+                              EventItem(event: events[idx]),
+                        )
+                      : Padding(
+                          padding: const EdgeInsets.all(8.0),
+                          child: Text(
+                            'At this moment, you are not joining any upcoming events. To find out what events are scheduled, check your spaces.',
```

Comment:
is this only showing _my_ events for this space? I'd argue this should probably show all upcoming events and their RSVP status...
```suggestion
                            'There are no events scheduled',
```